aws-android-sdk-cognitoauth/src/main/java/com/amazonaws/mobileconnectors/cognitoauth/AuthClient.java

```diff
@@ -709,21 +728,90 @@ private void launchSignOut(final String redirectUri, final String browserPackage
         launchCustomTabs(fqdn, null, browserPackage);
     }
 
+    /***
+     * Check if a browser is installed on the device to launch HostedUI.
+     * @return true if a browser exists else false.
+     */
+    private boolean isBrowserInstalled() {
+        if (isBrowserInstalled) {
+            return true;
```

Comment:
Looks like this caches the value if it is true, but not if it is false.  Can we cache it for false as well?  You could make it a `Boolean` instead of `boolean`, and check for null instead of checking if it's true.

Reply:
The value is set false by default. We should query for browsers every time in case the user installs a browsers after a failure. If we cache false value then the custom tabs won't launch even if the user just installed a browser. It will work though if user restarts the app.